rep/event.go

```diff
@@ -0,0 +1,33 @@
+package rep
+
+import (
+	"time"
+
+	"github.com/ipfs/go-cid"
+	"github.com/libp2p/go-libp2p-core/peer"
+)
+
+type RetrievalEventCode string
+
+const (
+	RetrievalEventConnect   RetrievalEventCode = "connect"
+	RetrievalEventQueryAsk  RetrievalEventCode = "query-ask"
+	RetrievalEventProposed  RetrievalEventCode = "proposed"
+	RetrievalEventAccepted  RetrievalEventCode = "accepted"
+	RetrievalEventFirstByte RetrievalEventCode = "first-byte-received"
+	RetrievalEventFailure   RetrievalEventCode = "failure"
+	RetrievalEventSuccess   RetrievalEventCode = "success"
+)
+
+type RetrievalEvent struct {
+	Code   RetrievalEventCode
+	Status string
+}
+
+type RetrievalState struct {
```

Comment:
It appears we aren't using this as the moment. Do we need to support it? Truthfully I think the main thing we will end up caring about is the PeerID, but I think we probably have most of these fields available whenever we publish a retrieval event, and I think we probably should include it. I guess it can be a separate PR though.

Reply:
Yeah, I think I was going to iterate off of this for the retrieval state metrics.